database/migrations/2023_09_09_195628_add_codex_tool.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('codex_text_types', function (Blueprint $table) {
+            $table->id();
+            $table->string('type');
+            $table->timestamps();
+        });
+
+        Schema::create('codex_texts', function (Blueprint $table) {
+            $table->id();
+            $table->integer('type_id')->unsigned();
```

Comment:
```suggestion
            $table->foreignId('codex_text_type_id')->contrained('codex_text_types');
```

Possibly add ->cascadeOnDelete()